interp/deprecation.ml

```diff
@@ -12,10 +12,12 @@ type t = { since : string option ; note : string option }
 
 let make ?since ?note () = { since ; note }
 
-let create_warning ~object_name ~warning_name name_printer =
+let create_warning ~object_name ~warning_name_if_no_since name_printer ?loc (qid,depr) =
   let open Pp in
-  CWarnings.create ~name:warning_name ~category:"deprecated"
+  let name = Option.cata (fun s -> "deprecated-since-" ^ s)
+    warning_name_if_no_since depr.since in
+  CWarnings.create ~name ~category:"deprecated"
```

Comment:
This is not the intended way of using CWarnings.create, it should be used once per warning not every time the warning is emitted.
It may work currently but if it does it is by chance.

Reply:
Right, the warning are now created only once.